fs/dafsserver/dafsserver.cpp

```diff
@@ -1396,21 +1461,370 @@ class CRemoteDiskReadActivity : public CRemoteDiskBaseActivity
     {
         return prefetchBuffer.tell() + baseFpos;
     }
-    virtual unsigned __int64 getLocalFilePosition(const void * row) override
+};
+
+
+class CRemoteCsvReadActivity : public CRemoteStreamReadBaseActivity
+{
+    typedef CRemoteStreamReadBaseActivity PARENT;
+
+    StringBuffer csvQuote, csvSeparate, csvTerminate, csvEscape;
+    unsigned __int64 headerLines = 0;
+    unsigned __int64 maxRowSize = 0;
+    bool preserveWhitespace = false;
+    CSVSplitter csvSplitter;
+    Owned<const IDynamicFieldValueFetcher> fieldFetcher;
+    Owned<const IDynamicTransform> translator;
+
+    class CFieldFetcher : public CSimpleInterfaceOf<IDynamicFieldValueFetcher>
+    {
+        CSVSplitter &csvSplitter;
```

Comment:
because of the way the splitter [potentially] 1st removes header lines, and how it's effectively used as a stream (eos) as well, I thought it made more sense to be used purely as a utility when dealing with the current line/fields.

Reply:
Leave for now.  When csv and disk read get refactored it will be obvious where the class needs to go.